tests/allocator-test.cc

```diff
@@ -491,8 +487,10 @@ namespace
 		 *  small sample of pointers.
 		 */
 		auto doAlloc = [&](HeapTestState &heap, size_t sz) {
-			CHERI::Capability p{heap_allocate(&t, heap.quota, sz)};
-
+			auto a = heap_allocate_cpp(&t, heap.quota, sz);
+			// TEST(!a.is_error(), "Fuzz allocation failed {}=>{}", sz,
+			// a.as_error());
```

Comment:
I don't _think_ we want allocation tests in the fuzzer to be a failure by default, but it might be nice to have it as a mode of running it when we know things about the heap and its other parameters?  (It's probably worth factoring the fuzzer out to its own `.cc` that can be built into the test harness or as a `tests.extra` or somesuch?)